int64 com
---
int64 check
